virtio-devices/src/balloon.rs

```diff
@@ -343,33 +343,26 @@ impl EpollHelperHandler for BalloonEpollHandler {
                     EpollHelperError::HandleEvent(anyhow!("Failed to get resize event: {:?}", e))
                 })?;
 
-                let mut signal_error = false;
-                let r = {
-                    let mut config = self.config.lock().unwrap();
-                    config.num_pages =
-                        (self.resize_receiver.get_size() >> VIRTIO_BALLOON_PFN_SHIFT) as u32;
-                    if let Err(e) = self.signal(VirtioInterruptType::Config) {
-                        signal_error = true;
-                        Err(e)
-                    } else {
-                        Ok(())
-                    }
-                };
-                let ret_err = anyhow!("{:?}", r);
-                if let Err(e) = &r {
-                    // This error will send back to resize caller.
-                    error!("Handle resize event get error: {:?}", e);
-                }
+                let mut config = self.config.lock().unwrap();
+                config.num_pages =
+                    (self.resize_receiver.get_size() >> VIRTIO_BALLOON_PFN_SHIFT) as u32;
+                self.signal(VirtioInterruptType::Config).map_err(|e| {
+                    let ret_err = anyhow!("Handle resize event get error: {:?}", e);
+                    self.resize_receiver
+                        .send(Err(e))
+                        .map_err(|e| {
+                            error!("Sending \"resize\" generated error: {:?}", e);
+                        })
+                        .ok();
```

Comment:
Here you're losing the error. Use `match` instead of `map_err()` so that you can propagate the error.

Reply:
This is intentional. My reasoning is that the `signal()` error is actual error that causes this function return in the first place, while the `receiver.send()` error will only happen from the error handling code (of the previous error). 

I think propagating the `signal()` error will help developer to understand the root cause of the failure. Let me know if I am mistaken.